sklearn/calibration.py

```diff
@@ -788,9 +743,8 @@ def predict_proba(self, X):
         proba : array, shape (n_samples, n_classes)
             The predicted probabilities. Can be exact zeros.
         """
+        predictions = _get_response_and_reshape(self.estimator, X)
```

Comment:
I assume that I would remove the function and directly call `_get_response_values` and just add

```python
if predictions.ndim == 1:
    predictions = prediction[:, np.newaxis]
```

Reply:
So that last failure (`test_setting_request_on_sub_estimator_removes_error`) fails because we use `ConsumingClassifier`, which will always output shape `(n_samples,)` (no matter the target type of `y`):

https://github.com/scikit-learn/scikit-learn/blob/24086ae24f747b3a4ba97af69b1461af1f760d8f/sklearn/tests/metadata_routing_common.py#L251-L255

I think this isn't a problem for any other metaestimator because only `CalibratedClassifierCV` will call a 'predict' method during `fit`. Do you think we need to do anything about this @glemaitre ?

Reply:
How do you decide between `reshape` and `newaxis` ? Having a quick look at: https://stackoverflow.com/questions/28385666/numpy-use-reshape-or-newaxis-to-add-dimensions I would lean towards reshape?

Reply:
I find `np.newaxis` more explicit that the `-1` and `1` meaning that might not be trivial to understand at first. However, if you are more comfortable with `reshape`, go with it. This is just a cosmetic and personal preference.